documentation/wasted_openAPI_specification.yaml

```diff
@@ -0,0 +1,377 @@
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '500':
+          $ref: '#/components/responses/ServerError'
+          
+  /challenge/{challengeId}:
+    parameters:
+      - in: path
+        name: challengeId
+        description: Challenge ID
+        required: true
+        schema:
+          type: integer
+          example: 1
+    get:
+      tags:
+        - Challenges
+      summary: Returns a list of challenge participant information based on challenge ID.
+      responses:
+        '200':
+          description: Success
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/UserChallenge'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/ServerError'
+  /challenge/{challengeId}/days:
+    parameters:
+      - in: path
+        name: challengeId
+        description: Challenge ID
+        required: true
+        schema:
+          type: integer
+          example: 1
+    get:
+      tags:
+        - Challenge Progress
+      summary: Returns a list of challenge days based on challenge ID.
+      responses:
+        '200':
+          description: Success
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/ChallengeDay'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/ServerError'
+  
+  /days/{dayId}:
+    parameters:
+      - in: path
+        name: dayId
+        description: Day ID
+        required: true
+        schema:
+          type: integer
+          example: 1     
+    put:
+      tags:
+        - Challenge Progress
+      summary: Modifies the progress status of a challenge day.
+      responses:
+        '204':
+          description: Edit successful
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/ChallengeDay'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/ServerError'
+
+components:
+  securitySchemes:
+    bearerAuth:           
+      type: http
+      scheme: bearer
+      bearerFormat: JWT   
+  schemas:
+    User:
+      properties:
```

Comment:
is_admin property is missing

Reply:
Does the API need to return this information (do we need it in the front end)? Or would it be for future implementations?

Reply:
I think you're right. We can leave it out